apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -1047,4 +1047,62 @@ mod test_prepare_proposal {
             assert_eq!(received, expected_txs);
         }
     }
+
+    /// Test if we are filtering out duped votes for the
+    /// same Ethereum event at different block heights.
+    #[cfg(not(feature = "abcipp"))]
+    #[test]
+    fn test_prepare_proposal_filter_duped_eth_event_votes() {
```

Comment:
This is allowed though (for the Tendermint v0.37 design)? We should be testing the opposite I think. Deduping (or rather ignoring of duplicate votes) will happen during `FinalizeBlock`.

Reply:
yeah this test was checking if the deduping logic was correct. I'm gonna purge that from this branch, so this test can also go along with it. it should still be adapted to finalize block though, I think.